### PR TITLE
Added C4 Charges

### PR DESCRIPTION
**Pull request by Michal78900**
Basically these are grenades that can stick to walls and be remotly detonated using `.detonate` command. Only the player who placed C4 can detonate it.

By default:
- C4s stick to walls
- Won't detonate themselves (unless someone will wait those 9999 seconds of fuse time, lol)
- When a player who have placed C4 charges dies (or leaves the game) his C4 charges will drop on the ground and someone other will be able to pick them up
- To detonate C4 charge, player need to have Detonator (by default it is a Radio) in hand

### CODE REVIEW

File: CustomItems/Items/C4Charge.cs [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+
+
+        /// <inheritdoc/>
+        protected override void SubscribeEvents()
+        {
+            Instance = this;
+
+            ServerEvent.WaitingForPlayers += OnWaitingForPlayers;
+
+            PlayerEvent.ThrowingGrenade += OnThrowing;
```
Comment (galaxy119) [inline CustomItems/Items/C4Charge.cs:131]: You're registering to an event that's already registered in the base method.

File: CustomItems/Items/C4Charge.cs [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+
+                    NetworkServer.Destroy(charge.Key.gameObject);
+                    PlacedCharges.Remove(charge.Key);
+                }
+            }
+        }
+
+        private void OnExploding(ExplodingGrenadeEventArgs ev)
+        {
+            if (PlacedCharges.ContainsKey(ev.Grenade.GetComponent<Grenade>()))
```
Comment (galaxy119) [inline CustomItems/Items/C4Charge.cs:221]: `
if (ev.Grenade.TryGetComponent(out Grenade grenade)
{
    PlacedCharges.Remove(grenade);
}
`

Comment (galaxy119) [reply CustomItems/Items/C4Charge.cs:221]: You shouldn't be calling GetComponent<> multiple times when you don't have to.

Also is there not a CustomGrenade virtual method for Exploding?

**Review by galaxy119 — CHANGES_REQUESTED** [review]

File: CustomItems/Items/C4Charge.cs (hunk 1 above)
Comment (Michal78900) [reply CustomItems/Items/C4Charge.cs:131]: Fixed

File: CustomItems/Items/C4Charge.cs (hunk 2 above)
Comment (Michal78900) [reply CustomItems/Items/C4Charge.cs]: Fixed component but I don't see the virtual method for grenade explode
image

Comment (galaxy119) [reply CustomItems/Items/C4Charge.cs]: You're right there isn't, I added it in 2.8.0-rc.1 but I'll handle updating the grenade items to use it.

**Review by galaxy119 — APPROVED** [review]
LGTM

**Pull request merged**